lib/puppet/util/windows/eventlog.rb

```diff
@@ -0,0 +1,114 @@
+require 'puppet/util/windows'
```

Comment:
@Iristyle yeah I think basically just redefining some of the FFI constants/methods, errors, and duplicate `wide_string` method would make it independent of puppet code. But then are you suggesting the same source file would somehow be leveraged by both daemon.rb and puppet, or just to copy the necessary code into daemon.rb? It seems to me it would be unusual to have daemon.rb use/know where eventlog.rb lives in puppet so I imagine the latter?